vsphere/internal/virtualdevice/virtual_machine_cdrom_subresource.go

```diff
@@ -389,30 +414,40 @@ func (r *CdromSubresource) Create(l object.VirtualDeviceList) ([]types.BaseVirtu
 	}
 
 	// We now have the controller on which we can create our device on.
-	dsID := r.Get("datastore_id").(string)
-	path := r.Get("path").(string)
-	ds, err := datastore.FromID(r.client, dsID)
-	if err != nil {
-		return nil, fmt.Errorf("cannot find datastore: %s", err)
-	}
-	dsProps, err := datastore.Properties(ds)
+	err = r.ValidateCdromConfig()
```

Comment:
`ValidateCdromConfig` should not be called here - this should have been done as part of a diff operation - so a function like `CdromDiffOperation` should have been created similar to the ones seen in disk and network device sub-resources. Each resource should be iterated over and run through the diff validation. This ensures that the validation is done at plan time and does not create confusion when someone attempts to do a create or update and runs into a configuration error they were told was okay before.